src/node/rpc/frontend.h

```diff
@@ -222,14 +224,32 @@ namespace ccf
             incoming_interface :
             interface_it.value().get<std::string>();
 
-          const auto primary_id = consensus->primary();
-          if (!primary_id.has_value())
+          std::optional<ccf::NodeId> target_node_id;
+          const auto nodes = InternalTablesAccess::get_trusted_nodes(tx);
+          {
+            const auto primary_id = consensus->primary();
```

Comment:
I was actually wondering if we should do the opposite, and deliberately return _a_ (randomly chosen) backup, rather than the first each time, for better work distribution.